### PR TITLE
fix gles build error

### PR DESCRIPTION
**Pull request by ijkmn**
1. gles 3.x also define GL_ES_VERSION_2_0
2. ICD don't export clGetGLContextInfoKHR, should get extension function address
3. remove ' include gl2extQCOM.h '

reopen PR of #651

### CODE REVIEW

File: test_common/gles/gl_headers.h [hunk 1]
```diff
@@ -40,21 +39,33 @@
...
+#endif
+
 #define glutGetProcAddress           eglGetProcAddress
 
 #define GL_FRAMEBUFFER_EXT           GL_FRAMEBUFFER
 #define GL_FRAMEBUFFER_COMPLETE_EXT  GL_FRAMEBUFFER_COMPLETE
 #define GL_RENDERBUFFER_INTERNAL_FORMAT_EXT GL_RENDERBUFFER_INTERNAL_FORMAT
 #define GL_RENDERBUFFER_EXT          GL_RENDERBUFFER
-#define GL_COLOR_ATTACHMENT0_EXT     GL_COLOR_ATTACHMENT0
+//#define GL_COLOR_ATTACHMENT0_EXT     GL_COLOR_ATTACHMENT0
```
Comment (kpet) [inline test_common/gles/gl_headers.h:64]: Why is this commented out?

File: test_common/gles/gl_headers.h [hunk 2]
```diff
@@ -40,21 +39,33 @@
...
+#define GL_RGBA32F GL_RGBA32F_EXT
+//#define GL_TEXTURE_3D                GL_TEXTURE_3D_OES
+#define GL_READ_ONLY GL_BUFFER_ACCESS_OES
+#define GL_HALF_FLOAT_ARB GL_HALF_FLOAT_OES
+#define GL_BGRA GL_BGRA_EXT
+#define glMapBufferRange glMapBufferRangeEXT
+#define glUnmapBuffer glUnmapBufferOES
+#else
+//#define GL_HALF_FLOAT_OES            GL_HALF_FLOAT
+//#define GL_RGBA8_OES                 GL_RGBA8
```
Comment (kpet) [inline test_common/gles/gl_headers.h:54]: Why are there commented-out definitions?

**Review by kpet — COMMENTED** [review]
This looks generally reasonable. A couple of questions.

File: test_common/gles/gl_headers.h (hunk 1 above)
Comment (ijkmn) [reply test_common/gles/gl_headers.h:64]: have defined in GLES2/gl2ext.h

File: test_common/gles/gl_headers.h (hunk 2 above)
Comment (ijkmn) [reply test_common/gles/gl_headers.h:54]: have defined in GLES2/gl2ext.h

**Comment by bcalidas** [comment]
This seems to be related to https://github.com/KhronosGroup/OpenCL-CTS/pull/976

976 includes a build fix for Android in addition to Windows.

File: test_common/gles/gl_headers.h (hunk 2 above)
Comment (gwawiork) [reply test_common/gles/gl_headers.h:54]: @ijkmn  I have build errors on ubuntu 20.04. Could you check on your side ?
test_fence_sync.cpp:300:24: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
  300 |         return (void *)error;

`
/home/gta/gwa/OpenCL-CTS/test_conformance/gles/test_fence_sync.cpp: In member function ‘virtual void* RunThread::IRun()’:
/home/gta/gwa/OpenCL-CTS/test_conformance/gles/test_fence_sync.cpp:300:24: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
  300 |         return (void *)error;
      |                        ^~~~~
In file included from /usr/include/string.h:495,
                 from /home/gta/gwa/OpenCL-CTS/test_conformance/gles/testBase.h:22,
                 from /home/gta/gwa/OpenCL-CTS/test_conformance/gles/test_fence_sync.cpp:18:
In function ‘char* strcpy(char*, const char*)’,
    inlined from ‘GLuint createShaderProgram(GLint*, GLint*)’ at /home/gta/gwa/OpenCL-CTS/test_conformance/gles/test_fence_sync.cpp:164:11:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:90:33: error: ‘void* __builtin___memcpy_chk(void*, const void*, long unsigned int, long unsigned int)’ writing 76 bytes into a region of size 75 overflows the destination [-Werror=stringop-overflow=]
   90 |   return __builtin___strcpy_chk (__dest, __src, __bos (__dest));
      |          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
`

Comment (ijkmn) [reply test_common/gles/gl_headers.h]: have fixed

**Review by gwawiork — APPROVED** [review]
LGTM

File: test_conformance/gles/CMakeLists.txt [hunk 3]
```diff
@@ -15,8 +15,8 @@ set (${MODULE_NAME}_SOURCES
         ../../test_common/gles/helpers.cpp
     )
 
-if(ANDROID)
-    list(APPEND CLConform_LIBRARIES GLESv2)
+if(ANDROID OR UNIX)
+    list(APPEND CLConform_LIBRARIES EGL GLESv2)
 elseif(WIN32)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DGLES3")
     list(APPEND CLConform_LIBRARIES libEGL libGLESv2 )
```
Comment (lakshmih) [inline test_conformance/gles/CMakeLists.txt:22]: Could we get rid of the "lib" prefix on the WIN32 case to match Android and Unix? (The build will still use "libLibrary.so" internally) This allows vendors to provide a CMake library object with one name via the CMakeVendors.txt file.

We have a windows development platform that compiles against GLESv2. To support both GLESv2 and GLESv3 on various implementations and to unify across various OSs, can we make the "-DGLES3" flag depend on a CMAKE cache variable?

**Comment by lakshmih** [comment]
We are okay to withdraw #976  and consolidate on this PR once the comments on the CMake file are addressed. This will ensure that we can build and run this test successfully on Android and on our development environment on Windows.

**Comment by kpet** [comment]
I believe https://github.com/KhronosGroup/OpenCL-CTS/pull/976 does everything this PR was doing. Closing. Feel free to re-open or create a new PR if that was not the case.

**Pull request closed without merging**